client/src/leap/soledad/client/api.py

```diff
@@ -70,6 +71,26 @@
 """
 SOLEDAD_CERT = None
 
+# A whitelist of modules from where to collect plugins dynamically.
+# For the moment restricted to leap namespace, but the idea is that we can pass
+# other "trusted" modules as options to the initialization of soledad.
+PLUGGABLE_LEAP_MODULES = ('mail', 'keymanager')
+
+
+def collect_plugins(interface):
```

Comment:
I believe the twisted plugin system relies on a cache that might be quite efficient, although memoizing might be appropiate as you say.

+1 to move to leap.common, I think I'll refactor it on the 2nd use of this.